libs/pipeline.py

```diff
@@ -0,0 +1,89 @@
+"""
+Code that is used to help move information around in the pipeline, starting with `Region` which
+represents a geographical area (state, county, metro area, etc).
+"""
+
+
+import json
+import os
+from typing import Optional
+
+import pandas as pd
+from pydantic import BaseModel
+
+import pyseir
+from covidactnow.datapublic.common_fields import CommonFields
+from libs.datasets import FIPSPopulation
+from libs.datasets.combined_datasets import load_us_latest_dataset, _log
+from pyseir.rt.utils import NEW_ORLEANS_FIPS
+from pyseir.utils import get_run_artifact_path, RunArtifact
+
+
+class Region(BaseModel):
+    """Identifies and provides access to data about a geographical area."""
+
+    # The FIPS identifier for the region, either 2 digits for a state or 5 digits for a county.
+    fips: str
+
+    def __repr__(self) -> str:
+        return f"Region(fips={self.fips})"
+
+    def get_us_latest(self):
+        """Gets latest values for a given state or county fips code."""
+        us_latest = load_us_latest_dataset()
+        return us_latest.get_record_for_fips(self.fips)
+
+    def load_inference_result(self):
+        """
+        Load fit results by state or county fips code.
+
+        Returns
+        -------
+        : dict
+            Dictionary of fit result information.
+        """
+        output_file = get_run_artifact_path(self.fips, RunArtifact.MLE_FIT_RESULT)
+        df = pd.read_json(output_file, dtype={"fips": "str"})
+        if len(self.fips) == 2:
+            return df.iloc[0].to_dict()
+        else:
+            return df.set_index("fips").loc[self.fips].to_dict()
+
+    def get_population(self) -> int:
```

Comment:
I do realize it's loading data, but this feels much cleaner as a property `population` rather than `get_population`.  Maybe the region object could load the `us_latest` on initialization or store the first call to `get_us_latest`?

Reply:
Done. I think my C++ and Java muscle memory is just more comfortable with methods than direct attribute access.